lib/views/reviews-view.js

```diff
@@ -0,0 +1,573 @@
+      return null;
+    }
+
+    const reviewAuthor = review.author ? review.author.login : '';
+    return (
+      <div className="github-ReviewSummary" key={review.id}>
+        <header className="github-ReviewSummary-header">
+          <span className={`github-ReviewSummary-icon icon ${icon}`} />
+          <img className="github-ReviewSummary-avatar"
+            src={review.author ? review.author.avatarUrl : ''} alt={reviewAuthor}
+          />
+          <a className="github-ReviewSummary-username" href={`https://github.com/${reviewAuthor}`}>{reviewAuthor}</a>
+          <span className="github-ReviewSummary-type">{copy}</span>
+          <Timeago className="github-ReviewSummary-timeAgo" time={review.submittedAt} displayStyle="short" />
+        </header>
+        <main className="github-ReviewSummary-comment">
+          <GithubDotcomMarkdown
+            html={review.bodyHTML}
+            switchToIssueish={this.props.openIssueish}
+            openIssueishLinkInNewTab={this.openIssueishLinkInNewTab}
+          />
+          <EmojiReactionsController
+            reactable={review}
+            tooltips={this.props.tooltips}
+            reportMutationErrors={this.props.reportMutationErrors}
+          />
+        </main>
+      </div>
+    );
+  }
+
+  renderReviewCommentThreads() {
+    const commentThreads = this.props.commentThreads;
+    if (commentThreads.length === 0) {
+      return null;
+    }
+
+    const resolvedThreads = commentThreads.filter(pair => pair.thread.isResolved).length;
+
+    const toggleComments = evt => {
+      evt.preventDefault();
+      if (this.props.commentSectionOpen) {
+        this.props.hideComments();
+      } else {
+        this.props.showComments();
+      }
+    };
+
+    return (
+      <details
+        className="github-Reviews-section comments"
+        open={this.props.commentSectionOpen}>
+
+        <summary className="github-Reviews-header" onClick={toggleComments}>
+          <span className="github-Reviews-title">Comments</span>
+          <span className="github-Reviews-progress">
+            <span className="github-Reviews-count">
+              Resolved
+              {' '}<span className="github-Reviews-countNr">{resolvedThreads}</span>{' '}
+              of
+              {' '}<span className="github-Reviews-countNr">{commentThreads.length}</span>
+            </span>
+            <progress className="github-Reviews-progessBar" value={resolvedThreads} max={commentThreads.length} />
+          </span>
+        </summary>
+        <main className="github-Reviews-container">
+          {commentThreads.map(this.renderReviewCommentThread)}
+        </main>
+
+      </details>
+    );
+  }
+
+  renderReviewCommentThread = commentThread => {
+    const {comments, thread} = commentThread;
+    const rootComment = comments[0];
+    if (!rootComment) {
+      return null;
+    }
+
+    let threadHolder = this.threadHolders.get(thread.id);
```

Comment:
it feels like this logic is getting a little out of hand -- we should refactor getting the line number and position text into a separate function.